File: src/java/medium/53. Maximum Subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int[] temp = new int[nums.length];
        int count = 0;
        for(int i = 0;i<nums.length;i++){
            count += nums[i];
            temp[i] = Math.max(nums[i],count);
            count = temp[i];
        }
        int result = temp[0];
        for(int i:temp){
            if(result < i){
                result = i;
            }
        }
        return result;
            
    }
}

File: src/golang/studyPlan/DataStructure/53. Maximum Subarray.go
func maxSubArray(nums []int) int {
    var tempResult = 0;
    var tempMap = make([]int, len(nums));
    
    for i, num := range nums{
        var temp = tempResult + num;
        if temp < num {
            tempMap[i] = num;
            
        } else {
            tempMap[i] = temp;
        }
        tempResult = tempMap[i];
    }
    var result = tempMap[0];
    for i:= 1;i<len(tempMap);i++{
        if(result < tempMap[i]){
            result = tempMap[i];
        }
    }
    return result;
}
